# Hi there ðŸ‘‹

<!--
**RichardDShipman/RichardDShipman** is a âœ¨ _special_ âœ¨ repository because its `README.md` (this file) appears on your GitHub profile.
-->

I am a bioinformatics consultant specializing in glycobiology with a background in applied science - concentration in biochemistry and molecular biology. Throughout my life, Iâ€™ve been tinkering with computers and the outdoors, which naturally led me to support the projects I pursue. Past experiences involved developing computational glyco / proteomics tools that enable the discovery and application of novel phosphopeptides and glycopeptides with desired properties.  I have a Master's degree in Omics Data Analysis from the University of Vic - Central University of Catalonia (UVic-UCC). I enjoy exploring and dabbling in the intersection of different omics fields with the aid of computational tools. As for recent projects, I am researching at the University of Valencia on the glycovirology of rotaviruses. The projects listed below highlight some of those adventures.

## Recent and Ongoing Projects

Here is a list of recent projects I have been toying with on my free time.

1. Glycopeptide Sequence Finder 
- A script for hunting (protease-digested) N/O/C-linked glycopeptides in protein/proteome FASTA files based on sequon (motif). It includes a zoo library of digested glycopeptides from various model organisms, common livestock & crops, sea critters and a host of fungus & viral pathogens. [Glycopeptide_Sequence_Finder](https://github.com/RichardDShipman/Glycopeptide_Sequence_Finder)
2. Glycoproteomics Graph Tool 
- Neo4j based graph knowledge base for storing glycoproteomics records in context to the central dogma of molecular biology. [Glycoproteomics_Graph_Tool](https://github.com/RichardDShipman/Glycoproteomics_Graph_Tool)
3.  Glycopeptide Proteoform Generator 
- A script to generate proteoforms by reading glycopeptide data from a CSV file, with limits on the number of proteoforms, saving results in both CSV and text file formats for each protein. [Glycopeptide_Proteoform_Generator](https://github.com/RichardDShipman/Glycopeptide_Proteoform_Generator)
4. Cross Species Pathway Analysis Pipeline
  - The Cross-Species Pathway Analysis Pipeline is an automated workflow designed to compare enzymatic pathways, such as the N-linked glycosylation pathway, across of species by integrating NCBI Datasets, BLAST+, and standard bioinformatics tools. [cross_species_pathway_analysis_pipeline](https://github.com/RichardDShipman/cross_species_pathway_analysis_pipeline)

## Past Projects

Here is a list of past projects I have worked on, with links to related GitHub repositories if they are available. 
1. Glycopeptide Data Analysis (Final Master Project - UVic)
- Comprehensive analysis of viral glycopeptides, including in-silico sequence prediction, conservation visualization via MSA, and experimental mass spectrometry data analysis. [Link](https://github.com/RichardDShipman/FMP_thesis_UVIC)
2.	Glycoproteomics Knowledge Database Development 
- Developed a Neo4j graph database for glycomics, integrating public multi-omics data.
3.	Glycoproteomics Mass Spectrometry Deep Learning Pipeline 
- Optimized a Python-based machine learning pipeline for glycoproteomics mass spectrometry, improving the accuracy of N-glycopeptide identification using deep learning models to analyze fragmentation patterns. [Link](https://github.com/Vennbiosciences/D-Va-GlycoML)
4. Dual-Ti-IMAC for Spike Protein Glycopeptide Analysis
- Developed a dual-functional Ti-IMAC method to characterize SARS-CoV-2 Spike protein glycosylation, identifying O-linked and M6P glycoforms. [Link](https://link.springer.com/article/10.1007/s00216-021-03433-1)
5. O-Linked Glycoproteomics of Alzhiemer's Disease Progression
- Monitored changes in the O-glycoproteome in the CSF of patients with progressing stages of AD. [LINK](https://pubmed.ncbi.nlm.nih.gov/34964596/)
6.	N-Linked Glycoform Categorization and Glycopeptide Data Analysis
- Developed R scripts for categorizing glycoforms based on mass spectrometry data, contributing to the analysis of glycosylation patterns in diseases. [Link](https://pubmed.ncbi.nlm.nih.gov/33862227/)

## Technical Skills

- Languages: Python, R, Bash
- Data Management: SQL, Neo4j - Cypher
- Data Visualization: R Shiny, Neodash, ggplot2
- Bioinformatics: Omics Data Analysis, Mass Spectrometry, Proteomics, Glycomics, Glycoproteomics

### LLMOps Projects

Couple of the starters projects I am working on for LLM agent use in computational biology research.

- TEXT_RAG [Link](https://github.com/RichardDShipman/TEXT_RAG) - A RAG that constructs a ChomaDB vector database for LLM querying of PDFs, HTMLs, and TXT files.

- csv_column_text_translator [Link](https://github.com/RichardDShipman/csv_column_text_translator) - Use LLMs to translate csv text column data to language of choice.