# Hi there üëã

<!--
**RichardDShipman/RichardDShipman** is a ‚ú® _special_ ‚ú® repository because its `README.md` (this file) appears on your GitHub profile.
-->

I am a bioinformatics consultant specializing in glycobiology with a background in applied science - concentration in biochemistry and molecular biology. Throughout my life, I‚Äôve been tinkering with computers and the outdoors, which naturally led me to support the projects I pursue. Past experiences involved developing computational glyco / proteomics tools that enable the discovery and application of novel phosphopeptides and glycopeptides with desired properties.  

I have a Master's degree in Omics Data Analysis from the University of Vic - Central University of Catalonia (UVic-UCC). I enjoy exploring and dabbling in the intersection of different omics fields with the aid of computational tools. As for recent projects, I am researching at the University of Valencia on the glycovirology of rotaviruses. The projects listed below highlight some of those adventures.

## Recent and Ongoing Projects

Here is a list of recent projects I have been toying with on my free time.

1. Glycopeptide Sequence Finder 
- A script for hunting (protease-digested) N/O/C-linked glycopeptides in protein/proteome FASTA files based on sequon (motif). It includes a zoo library of digested glycopeptides from various model organisms, common livestock & crops, sea critters and a host of fungus & viral pathogens. [Glycopeptide_Sequence_Finder](https://github.com/RichardDShipman/Glycopeptide_Sequence_Finder)
2. Glycoproteomics Graph Tool 
- Neo4j based graph knowledge base for storing glycoproteomics records in context to the central dogma of molecular biology. [Glycoproteomics_Graph_Tool](https://github.com/RichardDShipman/Glycoproteomics_Graph_Tool)
3.  Glycopeptide Proteoform Generator 
- A script to generate proteoforms by reading glycopeptide data from a CSV file, with limits on the number of proteoforms, saving results in both CSV and text file formats for each protein. [Glycopeptide_Proteoform_Generator](https://github.com/RichardDShipman/Glycopeptide_Proteoform_Generator)
4. Cross Species Pathway Analysis Pipeline
  - The Cross-Species Pathway Analysis Pipeline is an automated workflow designed to compare enzymatic pathways, such as the N-linked glycosylation pathway, across of species by integrating NCBI Datasets, BLAST+, and standard bioinformatics tools. [cross_species_pathway_analysis_pipeline](https://github.com/RichardDShipman/cross_species_pathway_analysis_pipeline)

## Past Projects

Here is a list of past projects I have worked on, with links to related GitHub repositories if they are available. 
1. Glycopeptide Data Analysis (Final Master Project - UVic)
- Comprehensive analysis of viral glycopeptides, including in-silico sequence prediction, conservation visualization via MSA, and experimental mass spectrometry data analysis. [Link](https://github.com/RichardDShipman/FMP_thesis_UVIC)
2.	Glycoproteomics Knowledge Database Development 
- Developed a Neo4j graph database for glycomics, integrating public multi-omics data.
3.	Glycoproteomics Mass Spectrometry Deep Learning Pipeline 
- Optimized a Python-based machine learning pipeline for glycoproteomics mass spectrometry, improving the accuracy of N-glycopeptide identification using deep learning models to analyze fragmentation patterns. [Link](https://github.com/Vennbiosciences/D-Va-GlycoML)
4. Dual-Ti-IMAC for Spike Protein Glycopeptide Analysis
- Developed a dual-functional Ti-IMAC method to characterize SARS-CoV-2 Spike protein glycosylation, identifying O-linked and M6P glycoforms. [Link](https://link.springer.com/article/10.1007/s00216-021-03433-1)
5. O-Linked Glycoproteomics of Alzheimer's Disease Progression
- Monitored changes in the O-glycoproteome in the CSF of patients with progressing stages of AD. [LINK](https://pubmed.ncbi.nlm.nih.gov/34964596/)
6.	N-Linked Glycoform Categorization and Glycopeptide Data Analysis
- Developed R scripts for categorizing glycoforms based on mass spectrometry data, contributing to the analysis of glycosylation patterns in diseases. [Link](https://pubmed.ncbi.nlm.nih.gov/33862227/)

## üõ†Ô∏è Technical Skills

* **Languages:** Python, R, Bash
* **Data Management:** SQL, Neo4j - Cypher
* **Data Visualization:** R Shiny, Neodash, ggplot2
* **Bioinformatics:** Omics Data Analysis, Computational Mass Spectrometry, Proteomics, Glycomics, Glycoproteomics

### LLMOps Projects

Couple of the starters projects I am working on for LLM agent use in computational biology research.

- TEXT_RAG [Link](https://github.com/RichardDShipman/TEXT_RAG) - A RAG that constructs a ChromaDB vector database for LLM querying of PDFs, HTMLs, and TXT files.

- csv_column_text_translator [Link](https://github.com/RichardDShipman/csv_column_text_translator) - Use LLMs to translate csv text column data to language of choice.

## üî¨ Publications & Presentations

### Peer-Reviewed Publications

1.  **[In-depth Site-Specific O-Glycosylation Analysis of Glycoproteins and Endogenous Peptides in Cerebrospinal Fluid (CSF) from Healthy Individual, Mild Cognitive Impairment (MCI) and Alzheimer‚Äôs disease (AD) patients](https://pubmed.ncbi.nlm.nih.gov/34964596/)**

      * Z. Chen, Q. Yu, J. Johnson, **R. Shipman**, X. Zhong, J. Huang, S. Asthana, C. Carlsson, O. Okonkwo, L. Li
      * *Published: December 2021*

2.  **[Simultaneous Enrichment and Separation of Neutral and Sialyl Glycopeptides of SARS-COV-2 Spike Protein Enabled by Dual-functionalized Ti-IMAC Material](https://link.springer.com/article/10.1007/s00216-021-03433-1)**

      * J. Huang, W. Danqing, **R. Shipman**, Z. Zexin, L. Yuan, L. Li
      * *Published: June 2021*

3.  **[In-depth site-specific analysis of N-glycoproteome in human cerebrospinal fluid (CSF) and glycosylation landscape changes in Alzheimer's disease (AD)](https://pubmed.ncbi.nlm.nih.gov/33862227/)**

      * Z. Chen, Q. Yu, Q. Yu, J. Johnson, **R. Shipman**, X. Zhong, J. Huang, S. Asthana, C. Carlsson, O. Okonkwo, L. Li
      * *Published: April 2021*

4.  **[Activity of Phosvitin in Hydroxyapatite Acid-Damage Immersion and Antimicrobial Assays](https://pmc.ncbi.nlm.nih.gov/articles/PMC7605931/)**

      * **R. Shipman**, S. Doering, J. Hemsath, E.J. Lee, J. E. Grant
      * *Published: October 2020*

### Conference Posters & Presentations

  * **Mapping the Glycosylation Landscape of the Rotavirus A Spike Protein Domain VP8\***

      * **R. Shipman**, S. L√≥pez-Navarro, R. C√°rcamo-Calvo, R. Gozalbo-Rovira, and J. Rodr√≠guez D√≠az
      * *8th Glycobasque Conference, San Sebastian, Spain 2025*

  * **A Property Graph Tool for Glycan Structural Information**

      * **R. Shipman**, K. Islas-Rios, N. Kitagawa, D. Serie, and M. P. Campbell
      * *6th Glycobasque Conference, Bilbao, Spain 2023*

  * **Targeting the Sugar Code: Serum Glycoproteome Profiling for Early Detection of Advanced Adenoma and Colorectal Cancer**

      * J. Liu, S. Grupta, A. Shaukat, N.S. Ding, H. Pohl, K. Desai, D. Chandraasekar, C. Guerrier, P. Ramachandran, K. Islas-Rios, **R. Shipman**, T. Caval, M. Campbell, C. Dhar, G. Xu, F. Schwarz, D. Serie, E. Dekker, M. Wallace, D. Hommes, F. May
      * *American College of Gastroenterology ACG 2023*

  * **Accurate Relative Glycoproteomic Quantitation in Complex Matrices (TMT, DDA-LFQ, PRM, and MS1 Filtering) Utilizing High-resolution Liquid Chromatography-Mass Spectrometry**

      * G. Gzerwieniec, I. Hundal, A. Antonopolis, R. Cheng, **R. Shipman**, X. Cong
      * *Glyco26 2023*

  * **D-Va: prediction and interpretation of intact N-glycopeptide tandem mass spectra by deep learning**

      * Z. Liang, M. P. Campell, **R. Shipman**, M. Gandhi, P. Aiyetan, N. Kitagawa, D. Serie
      * *American Society for Mass Spectrometry 2023*

  * **Evaluating a Glycan Database And Glycopeptide Result Converter Using Human Serum Across Multiple Glycopeptide Search Engines**

      * M. Gandhi, **R. Shipman**, M. P. Campell, N. Kitagawa, Z. Liang, G. Czerwieniec, X. Cong, D. Serie, K. Lindpaintner
      * *Society for Glycobiology 2022*

  * **An informatic approach to in-depth site-specific analysis of the glycoproteome in human cerebrospinal fluid and glycosylation alterations in Alzheimer's disease**

      * **R. Shipman**, Z. Chen, Q. Yu, J. Johnson, X. Zhong, J. Huang, S. Asthana, C. Carlsson, O. Okonkwo, L. Li
      * *North American Mass Spec Summer School 2020*

  * **Optimization of Protein Extraction and Identification from the European Green Crab**

      * N. Wrobel, G. Dachel, **R. Shipman**, A.D. Myracle, J.E. Grant
      * *Experiment Biology 2019 ‚Äì The American Society for Biochemistry and Molecular Biology*

  * **Modulation of gap junctional intercellular communication in HaCaT cells by xenobiotics**

      * **R. Shipman**, J. Jost, P. Waldmann, D. Pollet
      * *Experiment Biology 2019 ‚Äì The American Society for Biochemistry and Molecular Biology*

  * **In-depth site-specific analysis of glycoproteome in human cerebrospinal fluid (CSF) and glycosylation alterations in Alzheimer‚Äôs disease (AD)**

      * Z. Chen, Q. Yu, J. Johnson, **R. Shipman**, X. Zhong, J. Huang, S. Asthana, C. Carlsson, O. Okonkwo, L. Li
      * *American Society for Mass Spectrometry 2018*

  * **Development of an Immersion Model for Tooth Decay Observed via Scanning Electron Microscopy, Atomic Force Microscopy and Flame Atomic Absorption Spectroscopy**

      * **R. Shipman**, B. Quelle, J. Grant, E.J. Lee
      * *Experiment Biology 2018 ‚Äì The American Society for Biochemistry and Molecular Biology*

  * **In-depth site-specific analysis of glycoproteins and endogenous peptides in cerebrospinal fluid (CSF) from healthy individual, mild cognitive impairment (MCI) and Alzheimer‚Äôs disease (AD) patients**

      * Z. Chen, Q. Yu, J. Johnson, **R. Shipman**, X. Zhong, J. Huang, S. Asthana, C. Carlsson, O. Okonkwo, L. Li
      * *Wisconsin Alzheimer‚Äôs Disease & Related Disorders Research Day 2018*

  * **Effect of Metal Binding Proteins on the Surface of a Model of Tooth Decay**

      * **R. Shipman**, J. Grant, E. J. Lee
      * *Research in the Rotunda 2018 ‚Äì University of Wisconsin System*

  * **Assessing Student Competencies Using Rubrics Associated with Figure Legends and Ability to Perform the Bradford Assay in a 400-Level Undergraduate Proteomics Course**

      * J. Grant, **R. Shipman**
      * *Experiment Biology 2017 ‚Äì The American Society for Biochemistry and Molecular Biology*

  * **Effects of Phosvitin on Intestinal Epithelial Cell Viability Analyzed by Flow Cytometry and AlamarBlue Cell Viability Assay**

      * **R. Shipman**, J. Talukder
      * *Experiment Biology 2017 ‚Äì The American Society for Biochemistry and molecular Biology*

-----