## Hi there ðŸ‘‹

<!--
**RichardDShipman/RichardDShipman** is a âœ¨ _special_ âœ¨ repository because its `README.md` (this file) appears on your GitHub profile.
-->

I am a bioinformatician with a background in applied science, with a concentration in biochemistry and molecular biology. Throughout my life, Iâ€™ve been tinkering with computers, which naturally led me to support the projects I pursue. Past experiences involved developing computational proteomics tools that enable the discovery and application of novel phosphopeptides and glycopeptides with desired properties.  I am currently enrolled as a Master student in Omics Data Analysis at the University of Vic - Central University of Catalonia (UVic-UCC). I enjoy exploring the intersection of different omics fields with the aid of computational tools. The projects listed below highlight some of those adventures.

# Recent Projects

Here is a list of recent projects I have been toying with on my free time.

1. Glycopeptide Proteoform Generator (https://github.com/RichardDShipman/glycopeptide_proteoform_generator)
- A script to generate proteoforms by reading glycopeptide data from a CSV file, with limits on the number of proteoforms, saving results in both CSV and text file formats for each protein.
2. Glycoproteomics Graph Tool (https://github.com/RichardDShipman/GlycoproteomicsGraphTool_release)
- Neo4j based graph knowledge base for storing glycoproteomics records in context to the central dogma of molecular biology.
3. N-Glycopeptide Finder (https://github.com/RichardDShipman/N_Glycopeptide_Sequence_Finder)
- A script for (protease-digested) N-linked glycopetide hunting in proteome FASTA file input based on N-sequon.

# Past Projects

Here is a list of past projects I have worked on, with links to related GitHub repositories if they are available. 

1.	Glycoproteomics Knowledge Database Development 
- Developed a Neo4j graph database for glycomics, integrating public multi-omics data.
2.	Glycoproteomics Mass Spectrometry Deep Learning Pipeline (https://github.com/Vennbiosciences/D-Va-GlycoML)
- Optimized a Python-based machine learning pipeline for glycoproteomics mass spectrometry, improving the accuracy of N-glycopeptide identification using deep learning models to analyze fragmentation patterns.
3.	N-Linked Glycoform Categorization (10.1016/j.mcpro.2021.100081)
- Developed R scripts for categorizing glycoforms based on mass spectrometry data, contributing to the analysis of glycosylation patterns in diseases.
  
# Technical Skills

- Languages: Python, R, Bash
- Data Management: SQL, Neo4j - Cypher
- Data Visualization: R Shiny, Neodash, ggplot2
- Bioinformatics: Omics Data Analysis, Mass Spectrometry, Proteomics, Glycomics, Glycoproteomics
