## Hi there ðŸ‘‹

<!--
**RichardDShipman/RichardDShipman** is a âœ¨ _special_ âœ¨ repository because its `README.md` (this file) appears on your GitHub profile.
-->

My background is in applied science, with a concentration in biochemistry and molecular biology. Throughout my life, Iâ€™ve been tinkering with computers, which naturally led me to support the projects I pursue. Past experiences involved developing computational proteomics tools that enable the discovery and application of novel phosphopeptides and glycopeptides with desired properties.  I am currently enrolled as a Master student in Omics Data Analysis at the University of Vic - Central University of Catalonia (UVic-UCC). I enjoy exploring the intersection of different omics fields with the aid of computational tools. The projects listed below highlight some of those adventures.

# Past Projects

Here is a list of past projects I have worked on, with links to related GitHub repositories if they are available. 

1.	Glycoproteomics Knowledge Database Development (https://github.com/RichardDShipman/GlycoproteomicsGraphTool_release)
- Developed a Neo4j graph database for glycomics, integrating public and proprietary multi-omics data to enhance biomarker discovery for various disease indications.
2.	Data Visualization Tools
- Customized R Shiny and Neodash dashboards for improved visualization of glycoprotein data, facilitating accessibility and insights for data scientists.
3.	Glycoproteomics Mass Spectrometry Deep Learning Pipeline (https://github.com/Vennbiosciences/D-Va-GlycoML)
- Optimized a Python-based machine learning pipeline for glycoproteomics mass spectrometry, improving the accuracy of N-glycopeptide identification using deep learning models to analyze fragmentation patterns.
4.	Automated Glycopeptide Biomarker Identification
-	Automated a containerized pipeline for glycopeptide biomarker identification, streamlining the process for expanding marker panels.
5.	Comparative Proteomics Analysis
-	Conducted comparative proteomics analysis using tandem mass spectrometry to study specific proteins in various biological tissues(eye vs brain).
6.	Mitochondrial Dynamics Visualization
- Visualized mitochondrial dynamics in model organisms to monitor gene activity related to mitochondrial function and its impact on cellular processes.
7.	N-Linked Glycoform Categorization
- Developed R scripts for categorizing glycoforms based on mass spectrometry data, contributing to the analysis of glycosylation patterns in diseases.
  
# Technical Skills

- Languages: Python, R, Bash
- Data Management: SQL, Neo4j
- Data Visualization: R Shiny, Neodash, ggplot2
- Bioinformatics: Mass Spectrometry, Proteomics, Glycomics, Glycoproteomics
